core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/TypesafeDriverConfigProfile.java

```diff
@@ -72,6 +76,19 @@ public Duration getDuration(DriverOption option) {
     return getCached(option.getPath(), getEffectiveOptions()::getDuration);
   }
 
+  @Override
+  public Map<String, String> getMap(DriverOption option) {
+    Config subConfig = getCached(option.getPath(), getEffectiveOptions()::getConfig);
+    Map<String, String> map = new HashMap<>();
```

Comment:
Maybe I'm being paranoid, but I think it would be safer to return an immutable map. See `DefaultConsistencyLevel.mapByCode` for an example of how to build it.